compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/replacements/nodes/ArrayFillNode.java

```diff
@@ -0,0 +1,280 @@
+                    @ConstantNodeParameter EnumSet<?> runtimeCheckedCPUFeatures);
+
+    @Override
+    public ForeignCallDescriptor getForeignCallDescriptor() {
+        switch (this.elementKind) {
+            case Boolean:
+                return BOOLEAN_ARRAY_FILL_STUB;
+            case Byte:
+                return BYTE_ARRAY_FILL_STUB;
+            case Char:
+                return CHAR_ARRAY_FILL_STUB;
+            case Short:
+                return SHORT_ARRAY_FILL_STUB;
+            case Int:
+                return INT_ARRAY_FILL_STUB;
+            case Long:
+                return LONG_ARRAY_FILL_STUB;
+            case Float:
+                return FLOAT_ARRAY_FILL_STUB;
+            case Double:
+                return DOUBLE_ARRAY_FILL_STUB;
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public ValueNode[] getForeignCallArguments() {
+        return new ValueNode[]{arrayBase, offsetToFirstElement, arrayLength, valueToFillWith};
+    }
+
+    @Override
+    public void emitIntrinsic(NodeLIRBuilderTool gen) {
+        gen.getLIRGeneratorTool().emitArrayFill(elementKind, getRuntimeCheckedCPUFeatures(), gen.operand(arrayBase), gen.operand(offsetToFirstElement), gen.operand(arrayLength),
+                        gen.operand(valueToFillWith));
+    }
+
+    @Override
+    public LocationIdentity getLocationIdentity() {
+        return locationIdentity;
+    }
+
+    @Override
+    public boolean hasSideEffect() {
+        return true;
+    }
+
+    @Override
+    public LocationIdentity[] getKilledLocationIdentities() {
+        return new LocationIdentity[]{NamedLocationIdentity.getArrayLocation(this.elementKind)};
+    }
+
+    public static boolean isSupported(Architecture arch) {
+        return (arch instanceof AArch64);
+    }
+
+    @Override
+    public boolean canBeEmitted(Architecture arch) {
+        return isSupported(arch);
+    }
+
+    @Override
+    public FrameState stateAfter() {
+        return stateAfter;
+    }
+
+    @Override
+    public void setStateAfter(FrameState x) {
+        updateUsages(this.stateAfter, x);
+        this.stateAfter = x;
+    }
+
+    private static ForeignCallDescriptor createForeignCallDescriptor(JavaKind kind, Class<?> cls) {
+        LocationIdentity[] locs = new LocationIdentity[]{NamedLocationIdentity.getArrayLocation(kind)};
+        Class<?>[] argTypes = new Class<?>[]{Pointer.class, long.class, int.class, cls};
+        String name = kind.getJavaName() + "ArrayFill";
+
+        return new ForeignCallDescriptor(name, void.class, argTypes, HAS_SIDE_EFFECT, locs, false, false);
+    }
+
+    public JavaKind getElementKind() {
+        return elementKind;
+    }
+
+    public void setElementKind(JavaKind elementKind) {
+        this.elementKind = elementKind;
+    }
+
+    public ValueNode getArrayBase() {
+        return arrayBase;
+    }
+
+    public void setArrayBase(ValueNode arrayBase) {
+        this.arrayBase = arrayBase;
```

Comment:
Setters are not required for our Enterprise uses, so you could drop them. If you do include setters, note that setters for all inputs (fields marked `@Input`) must follow the pattern using `updateUsages`, as you did in `setStateAfter`.

On that note, I also noticed that you shouldn't need to override `hasSideEffect`, `stateAfter`, and `setStateAfter`. The implementations inherited from `AbstractStateSplit` should suffice.